### PR TITLE
[COMMS-572] Fix code block upcast crash on pasted rich text

### PR DESCRIPTION
**Pull request by thykel**
# Related PR
* https://github.com/opf/openproject/pull/24326

# Ticket
https://community.openproject.org/projects/COMMS/work_packages/COMMS-572/activity

# What are you trying to accomplish?
Pasting formatted rich text containing a syntax-highlighted code block crashed the editor.

`viewCodeBlockToModel` assumed `<code>`'s first child was a single text node and read `.data` off it, but highlight markup wraps the code in `<span>` elements, so `.data` was undefined and the uncaught error made the `EditorWatchdog` restart the editor, discarding unsaved content.

## Screenshots

# What approach did you choose and why?
Extract the text via a recursive `textContentOf()` helper that only reads `.data` off real text nodes, and cover it with unit tests (plain text, single text child, element-wrapped children, deep nesting, empty element).

# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: src/plugins/code-block/converters.js [hunk 1]
```diff
@@ -94,6 +99,28 @@ export function viewCodeBlockToModel() {
 }
 
 
+// Recursively collect the text of a view node, tolerating element
+// children. A pasted code block may contain syntax-highlight markup
+// (e.g. nested <span>s) rather than a single text node, in which case
+// reading `.data` off the first child yields `undefined` and throws.
```
Comment (brunopagno) [inline src/plugins/code-block/converters.js]: The `.data` part refers to a part of the code that is being removed in this PR. Not sure if this comment is really helping.

**Review by brunopagno — APPROVED** [review]
I tested locally, the changes make sense. So all good to me. Extra nice for specs 👍 

Only thing for me is the comments seem to be carrying too much context of the actual PR. I don't see much benefit on them.
